docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+    ```php
+    use OpenCloud\Rackspace;
+
+    $client = new Rackspace(Rackspace::US_IDENTITY_ENDPOINT, array(
+        'username' => '<YOUR RACKSPACE CLOUD ACCOUNT USERNAME>',
+        'apiKey'   => '<YOUR RACKSPACE CLOUD ACCOUNT API KEY>'
+     ));
+    ```
+
+### CDN service
+
+All CDN operations are done via an _CDN service object_. To
+instantiate this object, call the `cdnService` method on the `$client`
+object. This method takes one argument:
+
+| Position | Description | Data type | Required? | Default value | Example value |
+| -------- | ----------- | ----------| --------- | ------------- | ------------- |
+|  1       | Name of the service, as it appears in the service catalog | String | No | `null`; automatically determined when possible | `rackCDN` |
+
+
+```php
+$cdnService = $client->cdnService();
+```
+
+## Services
+
+A service represents your web application that has its content cached to the edge nodes.
+
+### Create a service (to represent your web application)
+
+
+This operation takes one parameter, an associative array, with the following keys:
+
+| Name | Description | Data type | Required? | Default value | Example value |
+| ---- | ----------- | --------- | --------- | ------------- | ------------- |
+| `name` | A human-readable name for the service. This name must be unique. | String | No | `null` | `acme_site` |
+| `domains` | An array of associative arrays, each specifying a domain name for your service. | Array of associative arrays | Yes | `null` | `array( array( 'domain' => 'www.acme.com' ), array ( 'domain' => 'acme.com' ) )` |
+| `origins` | An array of associative arrays, each specifying an origin server for your service. The `port`, `ssl`, and `rules` elements for each origin server are optional. | Array of associative arrays | Yes | `null` | `array( array( 'origin' => 'origin.acme.com', 'port' => 80, 'ssl' => false, 'rules' => array() ) )` |
+| `flavorId` | The flavor used to configure this service. Use the [list flavors](#list-flavors) operation to retrieve a list of all available flavors. | String | Yes| `null` | `cdn` |
+| `restrictions` | An array of associative arrays, each specifying a restriction for who can or cannot access content from the CDN cache. | Array of associative arrays | No | `null` | `array( array( 'name' => 'website only', 'rules' => array( array( 'name' => 'mywebsite.com', 'httpHost' => 'www.mywebsite.com' ) ) ) )` |
+| `caching` | An array of associative arrays, each specifying a caching rule for your service's assets. | No | `null` | `array( array( 'name' => 'default', 'ttl' => 3600 ) )` |
+
+You can create a service as shown in the following example:
+
+```php
+$service = $cdnService->createService(array(
+    'name'     => 'acme_site',
+    'domains'  => array(
+        array(
+            'domain' => 'www.acme.com'
+        ),
+        array(
+            'domain' => 'acme.com'
+        )
+    ),
+    'origins'  => array(
+        array(
+            'origin' => 'origin.acme.com'
```

Comment:
Can we contextualize a bit what `origin.acme.com` might actually look like? I'm used to the context of container caching - how might this be done with a subdomain? Would it access a filesystem?